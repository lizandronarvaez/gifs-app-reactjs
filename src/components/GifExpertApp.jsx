import { useState } from "react"
import AddCategory from "./AddCategory";
import GifCards from "./GifCards";
import Spinner from "./Spinner/Spinner";
const GifExpertApp = () => {

    // eslint-disable-next-line no-unused-vars
    const [categories, setCategories] = useState([]);

    const onAddCategory = (onCategoryNewValue) => {

        // Pasar el array de valores a lowercase
        const categoryExist = categories.map(cat => cat.toLowerCase())
        // Validar que el nombre existe
        if (categoryExist.includes(onCategoryNewValue)) {
            return;
        }
        // FORMAS DE LLAMAR AL STADO
        setCategories([onCategoryNewValue, ...categories])
        // setCategories(category=>[categories, ...category])
    }

    return (
        <>
            <div className="container">
                <h1>Gifs <span>Search</span> App</h1>
                <div className='form-submit'>
                    <AddCategory
                        // setCategories={setCategories}
                        // Con esta forma el hijo le emite un evento al padre
                        onCategoryNewValue={onAddCategory}
                    />
                    {
                        categories.length === 0 ?
                            <Spinner title={"Introduce un nombre para empezar la busqueda"} /> :
                            categories.map(category =>
                            (
                                <GifCards
                                    key={category}
                                    category={category}
                                />
                            )
                            )}
                </div>

            </div>

        </>
    )
}

export default GifExpertApp