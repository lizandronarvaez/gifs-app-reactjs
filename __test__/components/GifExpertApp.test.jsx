/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import { render, screen, renderHook } from "@testing-library/react";
import GifExpertApp from "../../src/components/GifExpertApp"
jest.mock("../../src/components/GifExpertApp");
describe('Pruebas en el componente <GifExpertApp />', () => {

    test('Hacer un snapshot al componente', () => {
        const { container } = render(<GifExpertApp />)

        expect(container).toMatchSnapshot();

        // screen.debug()
    });

    test('El estado inicial de la categoria es un array vacio', () => {

    });

    test('Solo se puede enviar una vez el mismo nombre de categoria', () => {

    });
});