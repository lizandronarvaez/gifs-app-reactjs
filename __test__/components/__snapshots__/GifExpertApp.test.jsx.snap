// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Hacer un snapshot al componente 1`] = `<div />`;
